[Android] ObjectDisposedException 'AndroidX.AppCompat.View.Menu.MenuItemImpl' Modifying Shell Items

### Description

I'm trying to dynamically load an app configuration file upon app launch and app resume from background which will define available Shell tabs and menu items.

This works reliably and consistently on iOS. On Android, however, it often crashes with the error log below on resume, and even sometimes on the initial launch.

The attached repro for simplicity allows the bug to happen by creating a new app shell upon resuming the app and populating it's tab bar and menu items within the app shells constructor.  Maintaining the same app shell instance and modifying the existing tabs/menu items instead often leads to the same crash.

### Steps to Reproduce

1. Download the attached Repro project.
2. Compile and run on an Android device.
3. Slowly and continuously send the app to the background and reopen.
4. The error should occur, however I only have one Android device to test with.


[Repro.zip](https://github.com/dotnet/maui/files/8917275/Repro.zip)



### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 10

### Did you find any workaround?

The same logic does work fine in Xamarin, but unfortunately no workaround in MAUI that I have been able to find yet.

### Relevant log output

```shell
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
[mono-rt] Object name: 'AndroidX.AppCompat.View.Menu.MenuItemImpl'.
[mono-rt]    at Java.Interop.JniPeerMembers.AssertSelf(IJavaPeerable self) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.cs:line 152
[mono-rt]    at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeAbstractObjectMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:line 373
[mono-rt]    at AndroidX.AppCompat.View.Menu.MenuItemImpl.SetIcon(Drawable icon) in C:\a\_work\1\s\generated\androidx.appcompat.appcompat\obj\Release
et6.0-android\generated\src\AndroidX.AppCompat.View.Menu.MenuItemImpl.cs:line 727
[mono-rt]    at Microsoft.Maui.Controls.Platform.BottomNavigationViewUtils.SetMenuItemIcon(IMenuItem menuItem, ImageSource source, IMauiContext context) in D:\a\_work\1\s\src\Controls\src\Core\Platform\Android\BottomNavigationViewUtils.cs:line 109
[mono-rt]    at Microsoft.Maui.Controls.Platform.BottomNavigationViewUtils.SetupMenu(IMenu menu, Int32 maxBottomItems, List`1 items, Int32 currentIndex, BottomNavigationView bottomView, IMauiContext mauiContext) in D:\a\_work\1\s\src\Controls\src\Core\Platform\Android\BottomNavigationViewUtils.cs:line 89
[mono-rt]    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
[mono-rt]    at Android.App.SyncContext.<>c__DisplayClass2_0.<Post>b__0() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:line 36
[mono-rt]    at Java.Lang.Thread.RunnableImplementor.Run() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:line 36
[mono-rt]    at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net6.0/android-32/mcw/Java.Lang.IRunnable.cs:line 84
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 22
```

--- a/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
+++ b/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
@@ -108,8 +108,8 @@ namespace Microsoft.Maui.Controls.Platform
 				source,
 				context.Context);
 
-			if (result is not null)
-				menuItem?.SetIcon(result.Value);
+			if (menuItem.IsAlive() && result is not null)
+				menuItem.SetIcon(result.Value);
 		}
 
 		public static BottomSheetDialog CreateMoreBottomSheet(

--- a/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
+++ b/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
@@ -94,7 +94,7 @@ namespace Microsoft.Maui.Controls.Platform
 
 		static async Task SetMenuItemIcon(IMenuItem menuItem, ImageSource source, IMauiContext context)
 		{
-			if (menuItem.IsDisposed())
+			if (!menuItem.IsAlive())
 				return;
 
 			if (source == null)

--- a/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
+++ b/src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs
@@ -94,6 +94,9 @@ namespace Microsoft.Maui.Controls.Platform
 
 		static async Task SetMenuItemIcon(IMenuItem menuItem, ImageSource source, IMauiContext context)
 		{
+			if (menuItem.IsDisposed())
+				return;
+
 			if (source == null)
 				return;
 
@@ -108,7 +111,6 @@ namespace Microsoft.Maui.Controls.Platform
 			if (result is not null)
 				menuItem?.SetIcon(result.Value);
 		}
-
 
 		public static BottomSheetDialog CreateMoreBottomSheet(
 			Action<int, BottomSheetDialog> selectCallback,

